Reduce JSC EWS false positives

kmiller68 opened the pull request:
#### c37cef3423f75c0c378eb148ec66dbf374cd1655
<pre>
Reduce JSC EWS false positives
<a href="https://bugs.webkit.org/show_bug.cgi?id=277428">https://bugs.webkit.org/show_bug.cgi?id=277428</a>
<a href="https://rdar.apple.com/132910943">rdar://132910943</a>

Reviewed by Aakash Jain, Yusuke Suzuki and Mark Lam.

Right now JSC EWS triggers when files such as JSCSSRuleCustom.cpp are modified because anything
with jsc in the path are considered relevant. In practice our `.*javascriptcore.*` will catch
anything in `Source/JavaScriptCore` and we only care about `.*jsc.*` when its inside `Tools/`

This also adds a pattern for `LayoutTests/js` since those tests run in the JSC harness too.

* Tools/CISupport/ews-build/steps.py:
(CheckChangeRelevance):

Canonical link: <a href="https://commits.webkit.org/281668@main">https://commits.webkit.org/281668@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e2efeb95af421668c2040b2d94f49be91b483090

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60590 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39950 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13166 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64521 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11137 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/62720 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47626 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11408 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/49015 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7740 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62623 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37217 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52491 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29843 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33908 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9720 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/10050 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55772 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/10021 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66250 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4534 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/9874 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/56385 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/60280 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4551 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52465 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/56558 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13504 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3758 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/35753 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/36836 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37929 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/36581 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

MenloDorian reviewed (approved):
r=me

briannafan commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -1249,13 +1249,13 @@ class CheckChangeRelevance(AnalyzeChange):
         re.compile(rb'Source/bmalloc/', re.IGNORECASE),
         re.compile(rb'Source/cmake/', re.IGNORECASE),
         re.compile(rb'.*Makefile.*', re.IGNORECASE),
+        re.compile(rb'Tools/.*jsc.*', re.IGNORECASE),
```
Nit: This could be `Tools/Scripts/.*jsc.*` to be clear about where the possible JSC scripts live.

kmiller68 replied:
I went with `Tools` because I figured it's possible some JSC related things could live outside `Tools/Scripts` and this would catch anything inside `Scripts`

kmiller68 committed:
Reduce JSC EWS false positives
https://bugs.webkit.org/show_bug.cgi?id=277428
rdar://132910943

Reviewed by Aakash Jain, Yusuke Suzuki and Mark Lam.

Right now JSC EWS triggers when files such as JSCSSRuleCustom.cpp are modified because anything
with jsc in the path are considered relevant. In practice our `.*javascriptcore.*` will catch
anything in `Source/JavaScriptCore` and we only care about `.*jsc.*` when its inside `Tools/`

This also adds a pattern for `LayoutTests/js` since those tests run in the JSC harness too.

* Tools/CISupport/ews-build/steps.py:
(CheckChangeRelevance):

Canonical link: https://commits.webkit.org/281668@main
